include/exadg/solvers_and_preconditioners/multigrid/multigrid_algorithm.h

```diff
@@ -167,25 +86,18 @@ class MultigridAlgorithm
         defect2[level] = solution[level];
     }
 
-#if ENABLE_TIMING
-    timings.init(minlevel, maxlevel);
-
-    for(unsigned int level = minlevel; level <= maxlevel; ++level)
-      timings.set(level, "dofs", defect[level].size());
-#endif
-  }
-
-  virtual ~MultigridAlgorithm()
-  {
-#if ENABLE_TIMING
-    timings.print(mpi_comm);
-#endif
+    timer_tree.reset(new TimerTree());
   }
 
   template<class OtherVectorType>
   void
   vmult(OtherVectorType & dst, OtherVectorType const & src) const
   {
+#if ENABLE_TIMING
+    Timer timer;
+    timer.restart();
```

Comment:
You don't need `restart()` after the constructor, as that already starts the timer.

Reply:
see newest commit